src/app/pages/actionable/components/actionable.active.view.component.ts

```diff
@@ -0,0 +1,220 @@
+import { FormGroup, FormControl, FormBuilder, AbstractControl, Validators, ReactiveFormsModule } from '@angular/forms';
+import { Observable } from 'rxjs/Rx';
+
+import { ActionableModel } from './actionable.model';
+import { ActionableService } from './actionable.service';
+import { ResponseModel, DataExchangeService, UtilityService } from '../../../shared';
+import { GlobalConstants } from '../../../shared/constants';
+
+@Component({
+    selector: 'actionable-active',
+    encapsulation: ViewEncapsulation.None,
+    templateUrl: '../views/actionable.active.html',
+    styleUrls: ['../styles/actionable.style.scss']
+})
+export class ActionableActiveComponent implements OnInit, OnDestroy, AfterContentInit {
+    @Input() DepartmentId: any;
+    @Input() IncidentId: any;
+    public form: FormGroup;
+    @ViewChild('myFileInput') myInputVariable: any;
+    private departmentId: number = null;
+    private incidentId: number = null;
+    editActionableModel: ActionableModel = null;
+    tempActionable: ActionableModel = null;
+    activeActionables: ActionableModel[] = [];
+    filesToUpload: Array<File>;
+    filepathWithLinks: string = null;
+    fileName: string = null;
+    constructor(formBuilder: FormBuilder, private actionableService: ActionableService,
+        private dataExchange: DataExchangeService<boolean>) {
+        this.filesToUpload = [];
+    }
+    ngOnInit(): any {
+        this.departmentId = Number(this.DepartmentId);
+        this.incidentId = Number(this.IncidentId);
+        this.getAllActiveActionable(this.incidentId, this.departmentId);
+        this.form = this.resetActionableForm();
+        this.dataExchange.Subscribe("OpenActionablePageInitiate", model => this.onOpenActionablePageInitiate(model));
+    }
+    private resetActionableForm(actionable?: ActionableModel): FormGroup {
+        return new FormGroup({
+            Comments: new FormControl(''),
+            URL: new FormControl('')
+        });
+    }
+    onOpenActionablePageInitiate(isOpen: boolean): void {
+        this.departmentId = Number(this.DepartmentId);
+        this.incidentId = Number(this.IncidentId);
+        this.getAllActiveActionable(this.incidentId, this.departmentId);
+    }
+    ngOnDestroy(): void {
+        this.dataExchange.Unsubscribe("OpenActionablePageInitiate");
+    }
+
+    ngAfterContentInit(): void {
+        this.setRagIntervalHandler();
+    }
+
+    IsDone(event: any, editedActionable: ActionableModel): void {
+        console.log(event.checked);
+        if (!event.checked) {
+            editedActionable.Done = false;
+        }
+        else {
+            editedActionable.Done = true;
+        }
+        let tempActionable = this.activeActionables.filter(function (item: ActionableModel) {
+            return (item.ActionId == editedActionable.ActionId);
+        });
+        tempActionable[0].Done = editedActionable.Done;
+    }
+    upload(actionableClicked: ActionableModel) {
```

Comment:
1. Move this functionality to the respective service
2. Use observable instead of promise for uploading.